src/modules/fw_pos_control_l1/fw_pos_control_l1_main.cpp

```diff
@@ -1616,6 +1630,111 @@ FixedwingPositionControl::control_position(const math::Vector<2> &current_positi
 				_global_pos.alt,
 				ground_speed,
 				tecs_status_s::TECS_MODE_NORMAL);
+
+	} else if (_control_mode.flag_control_offboard_enabled) {
+		/* OFFBOARD POSITION CONTROL */
+		double rel_lat, rel_lon;
+
+		if (_control_mode_current != FW_POSCTRL_MODE_OFFBOARD) {
+			/* Need to init because last loop iteration was in a different mode */
+			orb_copy(ORB_ID(home_position), _home_position_sub, &_home);
+			map_projection_init(&_home_ref, _home.lat, _home.lon);
+			_prev_offboard_position.valid = false;
+			_last_offboard_position.valid = false;
+		}
+		/* Reset integrators if switching to this mode from a other mode in which posctl was not active */
+		if (_control_mode_current == FW_POSCTRL_MODE_OTHER) {
+			/* reset integrators */
+			if (_mTecs.getEnabled()) {
+				_mTecs.resetIntegrators();
+				_mTecs.resetDerivatives(_airspeed.true_airspeed_m_s);
+			}
+		}
+		_control_mode_current = FW_POSCTRL_MODE_OFFBOARD;
+
+		/* current waypoint (the one we're currently heading for) */
+		math::Vector<2> curr_wp;
+		if (pos_sp_triplet.current.valid &&
+				pos_sp_triplet.current.position_valid) {
+			map_projection_reproject(&_home_ref,
```

Comment:
@LorenzMeier: Is it the task of the controllers to do all the projections? Aren't pos_sp_triplet.current.lat, etc. already provided?